anypoint-mq/4.x/modules/ROOT/pages/anypoint-mq-listener.adoc

```diff
@@ -129,23 +129,30 @@ You can acknowledge received messages automatically only if the flow execution f
 
 * *Automatic*
 +
-By default, the Subscriber source uses the AUTO acknowledgment mode. With this mode, the messages that the Subscriber source retrieves are acknowledged automatically after message flow processing succeeds. This means that the Subscriber source receives a message, dispatches it to the flow, and waits to see how the message processing finishes. It executes an ACK only when the processing finishes without exceptions.
+By default, the Subscriber source uses the `AUTO` acknowledgment mode. With this mode, the messages that the Subscriber source retrieves are acknowledged automatically after message flow processing succeeds. This means that the Subscriber source receives a message, dispatches it to the flow, and waits to see how the message processing finishes. It executes an ACK only when the processing finishes without exceptions.
 +
 If the execution of the processing flow finishes with a propagated exception, the message is automatically not acknowledged and is returned to the queue for redelivery.
 +
 For more information, see xref:anypoint-mq-ack.adoc#automatic-acknowledgment[Automatic Acknowledgment].
 
 * *Immediate*
 +
-When you use the IMMEDIATE acknowledgment mode, the consumed message is acknowledged right before being dispatched to the Mule flow. If the message acknowledgment fails, the message is discarded. The message isn't dispatched to the flow and remains in-flight until the acknowledgment timeout.
+When you use the `IMMEDIATE` acknowledgment mode, the consumed message is acknowledged (deleted from the queue) and then dispatched to the Mule flow for processing.
+If the message acknowledgment fails, the message is discarded. The message isn't dispatched to the flow and remains in-flight until the acknowledgment timeout.
```

Comment:
I think this note `If the app restarts between being acknowledged and dispatched, the message might not be available again because it was deleted from the queue.` will also make sense in this section?

Reply:
Yes, it's there on line 144-145.